components/admin_real/desktop/ReservationFieldsModal.jsx

```diff
@@ -0,0 +1,206 @@
+import { Alert, DatePicker, Input } from "antd";
+import { TimePicker } from "antd";
+import moment from "moment";
+import { Select } from "antd";
+import { dateFormat, timeFormat } from "../shared/constants";
+import { notification } from "antd";
+import FormField from "./FormField";
+
+const { Option } = Select;
+
+const ReservationFieldsModal = ({
+  reservation = {},
+  onSubmit,
+  modalTitle,
+  modalOpen = false,
+  setModalOpen,
+}) => {
+  const id = reservation.id || 123;
+
+  const [date, setDate] = useState(reservation.date);
+  const [time, setTime] = useState(reservation.time || 0);
+  const [certified, setCertified] = useState(
+    reservation.diver_certified ? reservation.diver_certified.toString() : null
+  );
+  const [reservationType, setReservationType] = useState(
+    reservation.reservation_type
+  );
+  const [name, setName] = useState(
+    reservation.diver_information ? reservation.diver_information.name : null
+  );
+  const [lastName, setLastName] = useState(
+    reservation.diver_information
+      ? reservation.diver_information.lastName
+      : null
+  );
+  const [age, setAge] = useState(
+    reservation.diver_information ? reservation.diver_information.age : null
+  );
+  const [experience, setExperience] = useState(reservation.number_of_dives);
+  const [email, setEmail] = useState(
+    reservation.diver_information ? reservation.diver_information.email : null
+  );
+
+  const openErrorNotification = (type) => {
+    notification[type]({
+      message: "Error",
+      description:
+        "There was an issue processing your request, please contact us if this persists. ",
+    });
+  };
+
+  const openSuccessNotification = (type) => {
+    notification[type]({
+      message: "Success",
+      description: "Request processed successfully!",
+    });
+  };
+
+  const reservationValid = () => {
+    if (
+      date &&
+      time &&
+      certified &&
+      reservationType &&
+      experience &&
+      name &&
+      lastName &&
+      age &&
+      email
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  return (
+    <Modal
+      title={modalTitle}
+      visible={modalOpen}
+      onOk={() => {
+        if (reservationValid()) {
+          onSubmit({
+            date,
+            time,
+            certified,
+            reservationType,
+            experience,
+            id,
+            diverInformation: {
+              name: name,
+              lastName: lastName,
+              age: age,
+              email: email,
+            },
+          }).then((results) =>
+            results.error
+              ? openErrorNotification("error")
+              : results.data
+              ? openSuccessNotification("success")
+              : console.log(results)
+          );
+          setModalOpen(false);
+        }
+      }}
+      onCancel={() => setModalOpen(false)}
+    >
+      <div className="flex flex-col">
+        {!reservationValid() && (
```

Comment:
Can you be more specific about the issue here? so I can fix it - I did not understand what you meant in your previous comment @adolfoportilla

Reply:
not sure what was my previous comment 😓